src/webgpu/api/operation/resource_init/check_texture/by_sampling.ts

```diff
@@ -79,19 +84,33 @@ export const checkContentsBySampling: CheckContents = (
     for (const slice of slices) {
       const ubo = t.device.createBuffer({
         mappedAtCreation: true,
-        size: 4,
+        size: 8,
         usage: GPUBufferUsage.UNIFORM | GPUBufferUsage.COPY_DST,
       });
-      new Int32Array(ubo.getMappedRange(), 0, 1)[0] = level;
+      new Int32Array(ubo.getMappedRange()).set([level, slice]);
       ubo.unmap();
 
       const byteLength =
         width * height * ReadbackTypedArray.BYTES_PER_ELEMENT * rep.componentOrder.length;
+
       const resultBuffer = t.device.createBuffer({
         size: byteLength,
         usage: GPUBufferUsage.STORAGE | GPUBufferUsage.COPY_SRC,
       });
 
+      const textureView =
+        params.dimension === '2d'
+          ? texture.createView({
+              baseMipLevel: 0,
+              mipLevelCount: params.mipLevelCount,
```

Comment:
is `mipLevelCount` the mipLevelCount of the`texture`? If so, this is the default (and I think it would be clearer to omit - right now, I can't tell whether this is supposed to be able to differ from the `texture`). In fact, all of these arguments in both branches are the defaults, except for `baseArrayLayer: slice`.